drivers/display/display_led_strip_matrix.c

```diff
@@ -0,0 +1,225 @@
+	const size_t offset = (mod_num_in_row * mod_row + mod_col) * mod_pixs;
+
+	if (config->circulative) {
+		if (config->start_from_right) {
+			mod_x = (config->module_width - 1) - mod_x;
+		}
+	} else {
+		if ((mod_y % 2) == !config->start_from_right) {
+			mod_x = (config->module_width - 1) - mod_x;
+		}
+	}
+
+	return offset + mod_y * config->module_width + mod_x;
+}
+
+static inline int check_descriptor(const struct led_strip_matrix_config *config, const uint16_t x,
+				   const uint16_t y, const struct display_buffer_descriptor *desc)
+{
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(desc->pitch <= config->width, "Pitch in descriptor is larger than screen size");
+	__ASSERT(desc->height <= config->height, "Height in descriptor is larger than screen size");
+	__ASSERT(x + desc->pitch <= config->width,
+		 "Writing outside screen boundaries in horizontal direction");
+	__ASSERT(y + desc->height <= config->height,
+		 "Writing outside screen boundaries in vertical direction");
+
+	if (desc->width > desc->pitch || x + desc->pitch > config->width ||
+	    y + desc->height > config->height) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int led_strip_matrix_write(const struct device *dev, const uint16_t x, const uint16_t y,
+				  const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	struct led_strip_matrix_data *data = dev->data;
+	const uint32_t *u32buf = buf;
+	int rc;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			uint32_t color = *u32buf++;
+			size_t idx = pixel_index(config, xpos, ypos);
+			struct led_rgb *pix = &data->pixels[idx];
+
+			pix->r = ((color >> 16) & 0xFF);
+			pix->g = ((color >> 8) & 0xFF);
+			pix->b = ((color) & 0xFF);
+		}
+		u32buf += (desc->pitch - desc->width);
+	}
+
+	rc = led_strip_update_rgb(config->dev, data->pixels, config->width * config->height);
+	if (rc) {
+		LOG_ERR("couldn't update strip: %d", rc);
+	}
+
+	return rc;
+}
+
+static int led_strip_matrix_read(const struct device *dev, const uint16_t x, const uint16_t y,
+				 const struct display_buffer_descriptor *desc, void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	struct led_strip_matrix_data *data = dev->data;
+	uint32_t *u32buf = buf;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			size_t idx = pixel_index(config, xpos, ypos);
+			struct led_rgb *pix = &data->pixels[idx];
+
+			*u32buf = (0xFF000000 | (pix->r << 16) | (pix->g << 8) | pix->b);
+			u32buf++;
+		}
+		u32buf += (desc->pitch - desc->width);
```

Comment:
same, no need for parenthesizing this

Reply:
:bookmark_tabs: Say no to overparenthisizing, say not to long overqualified identifiers, ... :smiley: